snapcraft/application.py

```diff
@@ -129,6 +129,11 @@ def __init__(self, *args, **kwargs):
         # Whether we know that we should use the core24-based codepath.
         self._known_core24 = False
 
+        try:
```

Comment:
can you add a code comment on why this is needed?

Reply:
Added comment explaining what the project path is used for.